src/lxc/conf.c

```diff
@@ -3506,17 +3522,12 @@ int lxc_map_ids(struct lxc_list *idmap, pid_t pid)
 		return -1;
 	}
 
-	for (type = ID_TYPE_UID; type <= ID_TYPE_GID; type++) {
-		int left, fill;
-		bool had_entry = false;
-		if (!buf) {
-			buf = pos = malloc(LXC_IDMAPLEN);
-			if (!buf)
-				return -ENOMEM;
-		}
-		pos = buf;
+	for (type = ID_TYPE_UID, u_or_g = 'u'; type <= ID_TYPE_GID;
+	     type++, type == ID_TYPE_GID ? (u_or_g = 'g') : (u_or_g = '@')) {
```

Comment:
If somehow the `enum` is altered I didn't want to just rewrite the gid mapping. But I think I just directly check that size(id types) == 2.